src/components/FileBrowser.vue

```diff
@@ -0,0 +1,285 @@
+        this.breadcrumbLoading = true;
+        const breadcrumb = { root: {}, path: [] };
+        let { id, type } = this.location;
+        while (type) {
+          const { data } = (await this.girderRest.get(`${type}/${id}`));
+          const entity = {
+            name: data._modelType !== 'user' ? data.name : data.login,
+            id: data._id,
+            type: data._modelType,
+            parentId: data.parentId,
+            parentType: data.parentCollection,
+          };
+          if (entity.type === 'folder') {
+            breadcrumb.path.unshift(entity);
+            id = entity.parentId;
+            type = entity.parentType;
+          } else {
+            breadcrumb.root = entity;
+            break;
+          }
+        }
+        this.breadcrumbLoading = false;
+        return breadcrumb;
+      },
+    },
+    counts: {
+      default: { nFolders: 0, nItems: 0 },
+      async get() {
+        const endpoint = `${this.location.type}/${this.location.id}/details`;
+        const { data } = (await this.girderRest.get(endpoint));
+        return {
+          nFolders: data.nFolders ? data.nFolders : 0,
+          nItems: data.nItems ? data.nItems : 0,
+        };
+      },
+      watch() {
+        return [
+          this.pagination,
+          this.location,
+        ];
+      },
+    },
+    rows: {
+      default: [],
+      async get() {
+        const request = async (gr, endpoint, params) => {
+          if (params === null) return [];
+          const resp = await gr.get(endpoint, { params });
+          if (resp.status !== 200) return [];
+          return resp.data;
+        };
+        this.rowsLoading = true;
+        const rows = [
+          ...(await request(this.girderRest, GIRDER_FOLDER_ENDPOINT, this.folderParams)),
+          ...(await request(this.girderRest, GIRDER_ITEM_ENDOINT, this.itemParams)),
+        ].map(item => ({
+          name: item.name,
+          type: item._modelType,
+          id: item._id,
+          size: item.size ? this.formatSize(item.size) : '',
+          icon: item._modelType in ICON_MAP ? ICON_MAP[item._modelType] : 'file',
+        }));
+        this.rowsLoading = false;
+        return rows;
+      },
+      watch() {
+        return [
+          this.folderParams,
+          this.itemParams,
+          this.counts,
+          this.location,
+          this.refreshCounter_,
+        ];
+      },
+    },
+  },
+  watch: {
+    location(newval, oldval) {
+      // force reset pagination when location changes.
+      if (newval.id !== oldval.id) {
+        this.pagination.page = 1;
+      }
+    },
+    selected(newval) {
+      this.$emit('selection-changed', newval);
+    },
+  },
+  computed: {
+    folderParams() {
+      return {
+        parentType: this.location.type,
+        parentId: this.location.id,
+        limit: this.pagination.rowsPerPage,
+        offset: (this.pagination.page - 1) * this.pagination.rowsPerPage,
+      };
+    },
+    itemParams() {
```

Comment:
Overall, the `itemParams` and `folderParams ` are only being used by the `rows` computed, in addition to the duplicate request behavior I mentioned in another comment, I am thinking maybe we should consider reconstruct the logic and make them simply local variables inside the rows computed, instead of two explicit computed.